small-talk/app/chat/page.js

```diff
@@ -4,17 +4,17 @@ import ThemeLayout from '../components/ThemeLayout';
 import socket from "../../util/socket";
 import React, { useEffect, useState } from 'react';
 import FriendsList from '../components/friends/FriendsList';
+import handleMessageSave from "@/db/messageSave"
 
 export default function Chat(){
 	const [user, setUser] = useState([]);
 	const [selectedUser, setSelectedUser] = useState(null);
 	const [privateMessages, setPrivateMessages] = useState([]);
 	const [inputMessage, setInputMessage] = useState("");
+	const [room, setRoom] = useState(""); 
 
 	useEffect(() => {
-		socket.auth = "random";
-		socket.connect();
-	
+		handleConnectButtonClick();
```

Comment:
This is great, we had to use the connect button before, but it is so much easier and nicer to have an automatic connection.